src/utils/sanitizeFileName.ts

```diff
@@ -2,12 +2,30 @@
 // eslint-disable-next-line no-control-regex
 const INVALID_CHAR_REGEX = /[\u0000-\u001F"#$&*+,:;<=>?[\]^`{|}\u007F]/g;
 const DRIVE_LETTER_REGEX = /^[a-z]:/i;
+const PERCENT_REGEX = /%/g;
+
+const decodeURIComponentSafe =
+	typeof decodeURIComponent === 'undefined' ? undefined : decodeURIComponent;
 
 export function sanitizeFileName(name: string): string {
+	// A `:` is only allowed as part of a windows drive letter (ex: C:\foo)
+	// Otherwise, avoid them because they can refer to NTFS alternate data streams.
 	const match = DRIVE_LETTER_REGEX.exec(name);
 	const driveLetter = match ? match[0] : '';
+	let path = name.slice(driveLetter.length).replace(INVALID_CHAR_REGEX, '_');
 
-	// A `:` is only allowed as part of a windows drive letter (ex: C:\foo)
-	// Otherwise, avoid them because they can refer to NTFS alternate data streams.
-	return driveLetter + name.slice(driveLetter.length).replace(INVALID_CHAR_REGEX, '_');
+	// Replace % with _ if there is invalid escape sequence
+	if (path.includes('%')) {
+		if (decodeURIComponentSafe) {
+			try {
+				decodeURIComponentSafe(path);
```

Comment:
Maybe you should have actually assigned the return value to `path`?

Reply:
I wrote reason why I didn't return the return value.

https://github.com/rollup/rollup/pull/5535#issuecomment-2141331133